jwst/photom/tests/test_photom.py

```diff
@@ -0,0 +1,1470 @@
+                slit.var_poisson = var_p
+                slit.var_rnoise = var_r
+                slit.var_flat = var_f
+                input_model.slits.append(slit.copy())
+        elif exptype == 'NRS_BRIGHTOBJ':
+            shape = (3, 5, 69)
+            (data, dq, err, var_p, var_r, var_f) = mk_data(shape)
+            wl = mk_wavelength(shape, 1.0, 5.0, dispaxis=1)
+            input_model = datamodels.SlitModel(data=data, dq=dq, err=err,
+                                               wavelength=wl)
+            input_model.var_poisson = var_p
+            input_model.var_rnoise = var_r
+            input_model.var_flat = var_f
+            input_model.name = 'S1600A1'
+        elif exptype == 'NRS_MSASPEC':
+            nslits = 3
+            input_model = datamodels.MultiSlitModel()
+            shape = (5, 69)
+            (data, dq, err, var_p, var_r, var_f) = mk_data(shape)
+            wl = mk_wavelength(shape, 1.0, 5.0, dispaxis=1)
+            for k in range(nslits):
+                slit = datamodels.SlitModel(data=data, dq=dq, err=err,
+                                            wavelength=wl)
+                slit.name = str(k + 1)
+                slit.var_poisson = var_p
+                slit.var_rnoise = var_r
+                slit.var_flat = var_f
+                input_model.slits.append(slit.copy())
+        else:
+            # This needs the wcs, so we won't cover this case.  Use a
+            # regression test instead.
+            raise RuntimeError("exp_type {} is not currently tested"
+                               .format(exptype))
+    elif instrument == 'NIRCAM':
+        if exptype == 'NRC_WFSS':
+            nslits = 1
+            input_model = datamodels.MultiSlitModel()
+            if pupil.endswith('C'):
+                shape = (69, 5)
+                dispaxis = 2                    # vertical
+            else:
+                shape = (5, 69)
+                dispaxis = 1                    # horizontal
+            (data, dq, err, var_p, var_r, var_f) = mk_data(shape)
+            wl = mk_wavelength(shape, 2.4, 5.0, dispaxis)
+            for k in range(nslits):
+                slit = datamodels.SlitModel(data=data, dq=dq, err=err,
+                                            wavelength=wl)
+                slit.name = str(k + 1)
+                slit.var_poisson = var_p
+                slit.var_rnoise = var_r
+                slit.var_flat = var_f
+                input_model.slits.append(slit.copy())
+        else:                                   # NRC_IMAGE
+            (data, dq, err, var_p, var_r, var_f) = mk_data((128, 256))
+            input_model = datamodels.ImageModel(data=data, dq=dq, err=err)
+            input_model.var_poisson = var_p
+            input_model.var_rnoise = var_r
+            input_model.var_flat = var_f
+    elif instrument == 'MIRI':
+        if exptype == 'MIR_MRS':
+            (data, dq, err, var_p, var_r, var_f) = mk_data((128, 256))
```

Comment:
Will the photom routines work OK on MIRI MRS data that isn't full-frame (you're creating a 128x256 subarray here)? If I recall correctly, one or more of the photom ref files for MRS mode is a full-frame image containing the flux calibration factors, hence it might not be possible to multiply it into a science data subarray.

Reply:
I wouldn't think so, no.  This only works because I created the "reference file" as well, and the function that creates it takes the image shape as an argument.